### PR TITLE
Add section on core workflow output SCE contents

### LINKED ISSUE #237: Add section on changes to SCE contents for core workflow
**Please provide some background on the proposed additions or changes.**

We currently don't describe anywhere what is now included in the `SingleCellExperiment` object that is output from the core workflow. 

**What are the changes that you are proposing?**

We should include a separate section in the expected output that talks about what is actually being added to the SCE object and how to access the changes. This should include things like a new `logcounts` assay, any additions to the `colData` and any additions to the `metadata` list. If this section gets long it may make sense to link out to an additional separate doc instead.

### PR DESCRIPTION
**Pull request by cbethell**
**Issue Addressed**
Closes #237 

**What is the purpose of these changes?**

This PR adds a subsection named `What to expect in the output SingleCellExperiment object` under the already existing `Expected Output` in the main `README.md` file.

**What changes did you make?**

In the added subsection, details around what is added to the SingleCellExperiment object and stored in the output .rds file are given.
More specifically, this includes information around the following:
- what is in the `logcounts` assay
- what is added to the `colData` of the SCE object, including default clustering results
- what is added to the `metadata` of the SCE object, including filtering type, cutoffs, and a note regarding normalization

**Any comments, concerns, or questions important for reviewers**
- Are the descriptions suffice for this use case?
- Do you have any suggestions on additional information that maybe should be added to this subsection?

**Checklist**

Place an `x` in all boxes that you have completed.

- [ ] I have run the most recent version of the code
- [ ] If I am adding in reports or generating any plots, I have reviewed the necessary reports
- [x] I have added all necessary documentation (if applicable)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -291,12 +291,30 @@ example_results
 	 └── <library_id>_<filtering_method>_processed_sce.rds
 ```
 
+The `<library_id>_<filtering_method>_core_analysis_report.html` file is the [html file](https://bookdown.org/yihui/rmarkdown/html-document.html#html-document) that contains the summary report of the filtering, dimensionality reduction, and clustering results associated with the processed `SingleCellExperiment` object.
```
Comment (allyhawkins) [inline README.md:294]:
```suggestion
The `<library_id>_core_analysis_report.html` file is the [html file](https://bookdown.org/yihui/rmarkdown/html-document.html#html-document) that contains the summary report of the filtering, dimensionality reduction, and clustering results associated with the processed `SingleCellExperiment` object.
```
Just as we are getting rid of the filtering method we mine as well do it here when you are making these changes too.

File: README.md [hunk 2]
```diff
@@ -291,12 +291,30 @@ example_results
...
+- The type of filtering performed (`miQC` or `manual`) on the expression data, which can be accessed using `metadata(sce)$filtering`.
+- The relevant filtering cutoffs.
+In the case of [`miQC`](https://bioconductor.org/packages/release/bioc/html/miQC.html) filtering, the maximum probability of cells being compromised is stored and can be accessed using `metadata(sce)$probability_compromised_cutoff`.
+See the [filtering parameters section](#filtering-parameters) for more information on what filtering cutoffs to expect here.
+- A note indicating whether or not clustering of similar cells passed the [`scran::quickCluster()`](https://rdrr.io/bioc/scran/man/quickCluster.html) step in the normalization script.
+If clustering of similar cells was successful, the note that reads "scater::logNormCounts clustered" can be found in `metadata(sce)$normalization`.
+- The subset of the most variable genes, determined using [`scran::getTopHVGs()`](https://rdrr.io/bioc/scran/man/getTopHVGs.html), can be accessed using `metadata(sce)$variable_genes`
```
Comment (allyhawkins) [inline README.md:315]: Can you turn this into a table instead? I think you could have a table with the name for each of the items in the metadata in one column and then a description. Then you have just one sentence stating how to access them using `metadata(sce)` .

**Review by allyhawkins — COMMENTED** [review]
The actual contents look good, I just had a comment on adding a table rather than having a bulleted list. Can you also make sure to update `doctoc`?

**Comment by cbethell** [comment]
Okay @allyhawkins, I've added a table like you suggested and updated doctoc. Let me know what you think!

File: README.md [hunk 3]
```diff
@@ -299,11 +300,29 @@ example_results
...
 
-The `<library_id>_core_analysis_report.html` file is the [html file](https://bookdown.org/yihui/rmarkdown/html-document.html#html-document) that contains the summary report of the filtering, dimensionality reduction, and clustering results associated with the processed `SingleCellExperiment` object.
+In the [`metadata`](https://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html#other-metadata) of the output `SingleCellExperiment` object, you can find the following information:
+
+| Metadata Information       | Description |
+|----------------------------|-------------|
+| Filtering method           | The type of filtering performed ([`miQC`](https://bioconductor.org/packages/release/bioc/html/miQC.html) or `manual`) on the expression data, which can be accessed using `metadata(sce)$filtering`. |
+| Filtering cutoffs          | The relevant filtering cutoffs. In the case of `miQC` filtering, the maximum probability of cells being compromised is stored and can be accessed using `metadata(sce)$probability_compromised_cutoff`. See the [filtering parameters section](#filtering-parameters) for more information on what filtering cutoffs to expect here. |
```
Comment (allyhawkins) [inline README.md:321]: I would actually list each of the filtering cutoffs as a separate row and then include a line with "Only present if filtering method used is `miQC`" and vice versa for all the manual filtering ones.

File: README.md [hunk 4]
```diff
@@ -299,11 +300,29 @@ example_results
...
+| Clustering of similar cells for normalization | A note indicating whether or not clustering of similar cells passed the [`scran::quickCluster()`](https://rdrr.io/bioc/scran/man/quickCluster.html) step in the normalization script. If clustering of similar cells was successful, the note that reads "scater::logNormCounts clustered" can be found in `metadata(sce)$normalization`. |
```
Comment (allyhawkins) [inline README.md:322]:
```suggestion
| `normalization` | Indicates if clustering of similar cells prior to normalization using [`scran::quickCluster()`](https://rdrr.io/bioc/scran/man/quickCluster.html) prior to normalization with `scater::logNormCounts()` was successful. |
```
Just simplifying this a little bit.

File: README.md [hunk 5]
```diff
@@ -299,11 +300,29 @@ example_results
...
+| Highly variable genes | The subset of the most variable genes, determined using [`scran::getTopHVGs()`](https://rdrr.io/bioc/scran/man/getTopHVGs.html), can be accessed using `metadata(sce)$variable_genes`. |
```
Comment (allyhawkins) [inline README.md:323]:
```suggestion
| `variable_genes` | The subset of the most variable genes, determined using [`scran::getTopHVGs()`](https://rdrr.io/bioc/scran/man/getTopHVGs.html). |
```

**Review by allyhawkins — COMMENTED** [review]
Just a little more formatting suggestions for the table. Can you make sure to explicitly list out all the contents of the metadata with the names used as I started doing for the ones you have so far?

**Comment by cbethell** [comment]
Those changes do look much better!! Let me know if you think I missed any important details however @allyhawkins

File: README.md [hunk 6]
```diff
@@ -299,11 +300,33 @@ example_results
...
+
+In the [`colData`](https://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html#handling-metadata) of the output `SingleCellExperiment` object, you can find the following:
+
+- Clustering results stored in a metadata column named using the associated clustering type and nearest neighbours values.
 For example, if using the default values of Louvain clustering with a nearest neighbors parameter of 10, the column name would be `louvain_10` and can be accessed using `colData(sce)$louvain_10`.
 
-The `<library_id>_core_analysis_report.html` file is the [html file](https://bookdown.org/yihui/rmarkdown/html-document.html#html-document) that contains the summary report of the filtering, dimensionality reduction, and clustering results associated with the processed `SingleCellExperiment` object.
+In the [`metadata`](https://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html#other-metadata) of the output `SingleCellExperiment` object, which can be accessed using `metadata(sce)` you can find the following information:
```
Comment (allyhawkins) [inline README.md:316]:
```suggestion
In the [`metadata`](https://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html#other-metadata) of the output `SingleCellExperiment` object, which can be accessed using `metadata(sce)`,  you can find the following information:
```
Adding missing comma.

File: README.md [hunk 7]
```diff
@@ -299,11 +300,33 @@ example_results
...
+
+| Metadata Information       | Description |
```
Comment (allyhawkins) [inline README.md:318]:
```suggestion
| Metadata Key       | Description |
```

File: README.md [hunk 8]
```diff
@@ -299,11 +300,33 @@ example_results
...
+|----------------------------|-------------|
+| `filtering_method`           | The type of filtering performed ([`miQC`](https://bioconductor.org/packages/release/bioc/html/miQC.html) or `manual`) on the expression data. |
+| `prob_compromised_cutoff` | The maximum probability of cells being compromised, which only present when the `filtering_method` is set to `miQC`. |
```
Comment (allyhawkins) [inline README.md:321]:
```suggestion
| `prob_compromised_cutoff` | The maximum probability of cells being compromised, which is only present when the `filtering_method` is set to `miQC`. |
```

File: README.md [hunk 9]
```diff
@@ -299,11 +300,33 @@ example_results
...
+| `miQC_model` | The linear mixture model calculated by `miQC` and therefore is only present when `filtering_method` is set to `miQC`. |
+| `mito_percent_cutoff` | Maximum percent mitochondrial reads per cell threshold, which is only present when `filtering_method` is set to `manual`. |
+| `detected_gene_cutoff` | Minimum number of genes detected per cell, which is only present when `filtering_method` is set to `manual`. |
+| `umi_count_cutoff` | Minimum unique molecular identifiers (UMI) per cell, which is only present when `filtering_method` is set to `manual`. |
+| `normalization` | Indicates if clustering of similar cells prior to normalization using [`scran::quickCluster()`](https://rdrr.io/bioc/scran/man/quickCluster.html) prior to normalization with `scater::logNormCounts()` was successful. |
```
Comment (allyhawkins) [inline README.md:326]:
```suggestion
| `normalization` | Indicates if clustering of similar cells using [`scran::quickCluster()`](https://rdrr.io/bioc/scran/man/quickCluster.html) prior to normalization with `scater::logNormCounts()` was successful. |
```

**Review by allyhawkins — APPROVED** [review]
Just some minor word edits, but I don't need to see it again. Looks good!

**Pull request merged**